crates/forge/tests/e2e/workspaces.rs

```diff
@@ -0,0 +1,306 @@
+        [PASS] hello_workspaces::test_simple
+        Running 2 test(s) from tests/test_failing.cairo
+        [FAIL] test_failing::test_failing
+        
+        Failure data:
+            original value: [8111420071579136082810415440747], converted to a string: [failing check]
+        
+        [SKIP] test_failing::test_another_failing
+        Tests: 1 passed, 1 failed, 1 skipped
+        "#});
+}
+
+#[test]
+fn run_inside_nested_package() {
+    let temp = assert_fs::TempDir::new().unwrap();
+    temp.copy_from("tests/data/hello_workspaces", &["**/*.cairo", "**/*.toml"])
+        .unwrap();
+
+    let package_dir = temp.join(PathBuf::from("crates/addition"));
+
+    let snapbox = runner();
+
+    snapbox
+        .current_dir(package_dir)
+        .assert()
+        .success()
+        .stdout_matches(indoc! {r#"
+        [..]Compiling[..]
+        [..]Finished[..]
+        Collected 4 test(s) and 3 test file(s)
+        Running 1 test(s) from addition package
+        [PASS] addition::tests::it_works
+        Running 2 test(s) from tests/nested/test_nested.cairo
+        [PASS] test_nested::test_two
+        [PASS] test_nested::test_two_and_two
+        Running 1 test(s) from tests/test_simple.cairo
+        [PASS] test_simple::simple_case
+        Tests: 4 passed, 0 failed, 0 skipped
+        "#});
+}
+
+#[test]
+fn run_for_entire_workspace() {
+    let temp = assert_fs::TempDir::new().unwrap();
+    temp.copy_from("tests/data/hello_workspaces", &["**/*.cairo", "**/*.toml"])
+        .unwrap();
+
+    let snapbox = runner().arg("--workspace");
+
+    snapbox
+        .current_dir(&temp)
+        .assert()
+        .success()
+        .stdout_matches(indoc! {r#"
+        [..]Compiling[..]
+        [..]Finished[..]
+        Collected 4 test(s) and 3 test file(s)
+        Running 1 test(s) from addition package
+        [PASS] addition::tests::it_works
+        Running 2 test(s) from tests/nested/test_nested.cairo
+        [PASS] test_nested::test_two
+        [PASS] test_nested::test_two_and_two
+        Running 1 test(s) from tests/test_simple.cairo
+        [PASS] test_simple::simple_case
+        Tests: 4 passed, 0 failed, 0 skipped
+        Collected 1 test(s) and 1 test file(s)
+        Running 1 test(s) from fibonacci package
+        [PASS] fibonacci::tests::it_works
+        Tests: 1 passed, 0 failed, 0 skipped
+        Collected 3 test(s) and 2 test file(s)
+        Running 1 test(s) from hello_workspaces package
+        [PASS] hello_workspaces::test_simple
+        Running 2 test(s) from tests/test_failing.cairo
+        [FAIL] test_failing::test_failing
+        
+        Failure data:
+            original value: [8111420071579136082810415440747], converted to a string: [failing check]
+        
+        [SKIP] test_failing::test_another_failing
+        Tests: 1 passed, 1 failed, 1 skipped
+        "#});
+}
+
+#[test]
+fn run_for_entire_workspace_and_specific_package() {
```

Comment:
I think we should have an error about arguments being exclusive here. I understand that it is not how cargo works, but in this case I don't think it useful to mimic its behavior.

cc @piotmag769

Reply:
I'm fine with that if it suits you better